Python/crossinterp.c

```diff
@@ -713,6 +713,69 @@ _bool_shared(PyThreadState *tstate, PyObject *obj,
     return 0;
 }
 
+struct _shared_tuple_data {
+    Py_ssize_t len;
+    _PyCrossInterpreterData ** data;
+};
+
+static PyObject *
+_new_tuple_object(_PyCrossInterpreterData *data)
+{
+    struct _shared_tuple_data *shared = (struct _shared_tuple_data *)(data->data);
+    PyObject *tuple = PyTuple_New(shared->len);
+
+    for (Py_ssize_t i = 0; i < shared->len; i++) {
+        PyObject *item = _PyCrossInterpreterData_NewObject(shared->data[i]);
+        PyTuple_SET_ITEM(tuple, i, item);
```

Comment:
In review please confirm this is correct and doesn't require an additional incref? 
The objects should have a ref count of 1 I think once the tuple is made so if the tuple is destroyed so are the items.

Reply:
Correct.  `_PyCrossInterpreterData_NewObject()` returns a new reference, which `PyTuple_SET_ITEM()` steals[^1], so it balances out just right.  Neither an additional incref nor an additional decref are required.

[^1]: See the note at https://docs.python.org/3.12/c-api/tuple.html?highlight=pytuple_set_item#c.PyTuple_SET_ITEM.